source/iodhbwm-templates.sty

```diff
@@ -84,9 +84,20 @@
 \def\iodhbwm@bachelor@type@bos{Bachelor of Science}
 \def\iodhbwm@bachelor@type@boa{Bachelor of Arts}
 
-\def\iodhbwm@thesis@type@ba{Bachelorarbeit}
-\def\iodhbwm@thesis@type@pa{Projektarbeit}
-\def\iodhbwm@thesis@type@sa{Studienarbeit}
+\def\iodhbwm@thesis@type@ba{UNDEFINED LANGUAGE}
```

Comment:
If I get this right, there is no default language set at the moment. What would happen now if a user select spanish or whatever as a language?
I would prefer to set German as default or fallback solution and print a warning if there is no template available for the selected language.

Reply:
That's correct currently only english and ngerman are supported. With any other language, it kinda dies right away at compilation. 
But don't overestimate my latex skills. You wanna add this by yourself?

Reply:
To be honest, I don't have enought time anymore to add new features. It would be enought if there will be a warning and the german template is used.
So maybe `\iodhbwm@main@language` can be set to ngerman by default and a internal list with supported languages is provided. Let me see if I will find some time within the next few days because I think this is a nice feature :)

Reply:
It sounds like you have a pretty clear idea in your head, how this should work and i see the benefits of defaulting to a language with an warning.

Can you describe your idea a bit more, and I'll try to do it myself

Reply:
Okay lets give a try:

1. By default all external files are loaded from `/ngerman/`
2. The I would provide a new files called `i18n.def`
    - In this file all _supported_ languages are listed with a `\iflanguage`
    - Inside the condition all necessary variable will get overridden

So these steps will ensure that the class can always compile.

In addition it would be nice if somehow a kind of list with supported languages is provided. At this point we can check if the selected languages is supported otherwise a latex warning is printed in the console and the default language (German) is used.

For me it is mantatory that the class will almost always compile and print some sort of result.

Reply:
So it kind of has the wanted behavior now.
Running it with spanish works, it will print the german template, but chooses spanish, e.g. for the directory names.

What breaks the class if one language is not implemented correctly. E.g., if spanish is defined in `i18n.def`, but the respective `.def` files are missing, the compilation will fail.

But this error will probably only occur to devs and not to users, so I think it's okay.

At line 363 and following, there are already some precautions taken to prevent the class from failing by wrong user input.

Reply:
Thanks a lot for your help. I just did some minor changes to your code.